### ISSUE TITLE
sql: error decoding JSON: invalid header

### ISSUE THREAD

**Issue by rohany** [cockroachdb/cockroach#51475]
See these error logs: sqlsmith/setup=seed/setting=no-ddl/ -- 

Query errors out:
```
20:41:24 test.go:325: test failure: 	sqlsmith.go:169,sqlsmith.go:199,test_runner.go:757: error: pq: internal error: error decoding JSON value, header: 3be200
		stmt:
		SELECT
			(
				SELECT
					tab_651._string AS col_1646
				FROM
					defaultdb.public.seed@seed__int8__float8__date_idx AS tab_650
					JOIN defaultdb.public.seed@[0] AS tab_651 ON
							(tab_650._float4) = (tab_651._float4)
							AND (tab_650._int4) = (tab_651._int4)
							AND (tab_650._date) = (tab_651._date)
							AND (tab_650._bool) = (tab_651._bool)
							AND (tab_650._int2) = (tab_651._int8),
					defaultdb.public.seed AS tab_652
					JOIN defaultdb.public.seed AS tab_653 ON (tab_652._jsonb) = (tab_653._jsonb)
					JOIN defaultdb.public.seed@seed__int8__float8__date_idx AS tab_654 ON (tab_652._jsonb) = (tab_654._jsonb)
					JOIN defaultdb.public.seed@seed__int8__float8__date_idx AS tab_655 ON (tab_654._int2) = (tab_655._int4)
				ORDER BY
					tab_655._bool DESC
				LIMIT
					1:::INT8
			)
				AS col_1647
		LIMIT
			67:::INT8;
```

**Comment by solongordon** [comment]
Let's investigate severity to make a call on whether we need a fix for 20.2.

**Comment by jordanlewis** [comment]
Old/not reproable
